lib/arithmetic.js

```diff
@@ -159,12 +228,22 @@ exports.traceExpression = function(builder, expression, traceOpts) {
   // expand any globbing operations in the expression
   var expandPromise = Promise.resolve();
   operations.forEach(function(operation) {
-    expandPromise = expandPromise.then(function() {
-      return Promise.resolve(expandGlobAndCanonicalize(builder, operation))
-      .then(function(expanded) {
-        expandedOperations = expandedOperations.concat(expanded);
+    if (operation.bulkOperation) {
+      var expandedTreePromise = traceExpression(builder, operation.bulkOperation, traceOpts)
+      expandPromise = expandPromise.then(function() {
+        return Promise.resolve(expandedTreePromise)
+               .then(function(fullTree){
+                 expandedOperations = expandedOperations.concat({ preExpandedTree: fullTree, operator: operation.operator })
```

Comment:
Can this still be called `bulkOperation` instead of `preExpandedTree` or am I missing something here?